tribits/core/package_arch/TribitsExternalPackageFindTplHelpers.cmake

```diff
@@ -108,35 +125,74 @@ function(tribits_extpkg_create_package_config_file_with_imported_targets
+
+  # Get ${externalPkg} from where you found it before (see note below)
+  tribits_extpkg_append_find_dependency_external_package_with_imported_targets(
+    ${tplName}  ${externalPkg}  configFileStr)
+
+  # Pull in upstream <UpstreamPkg>Config.cmake files
+  tribits_extpkg_add_find_upstream_dependencies_str(${tplName} configFileStr)
+
+  # Add the ${tplName}::all_libs target and link to this ${externalPkg}
+  # package's native IMPORTED targets
+  tribits_extpkg_create_all_libs_target( ${tplName}
+    LIB_TARGETS_LIST  ${PARSE_IMPORTED_TARGETS_FOR_ALL_LIBS}
+    CONFIG_FILE_STR_INOUT  configFileStr )
+
+  # Also link against upstream package's `<UpstreamPkg>::all_libs` targets
+  tribits_extpkg_append_upstream_target_link_libraries_str(${tplName}
+    ${tplName}::all_libs  configFileStr)
+
+  tribits_append_tribits_compliant_package_config_vars(${tplName}  configFileStr)
+
+  tribits_extpkg_write_package_config_file_from_str(${tplName}  "${configFileStr}")
+endfunction()
+
+
+function(tribits_extpkg_append_find_dependency_external_package_with_imported_targets
+    tplName  externalPkg  configFileStrVarInOut
+  )
+  set(configFileStr "${${configFileStrVarInOut}}")
   if (${externalPkg}_DIR)
     string(APPEND configFileStr
       "set(${externalPkg}_DIR \"${${externalPkg}_DIR}\")\n" )
   endif()
   string(APPEND configFileStr
-    "find_dependency(${externalPkg})\n"
-    "add_library(${tplName}::all_libs  INTERFACE  IMPORTED  GLOBAL)\n"
-    )
-  foreach (importedTarget  IN LISTS  PARSE_IMPORTED_TARGETS_FOR_ALL_LIBS)
-    string(APPEND configFileStr
-      "target_link_libraries(${tplName}::all_libs  INTERFACE  ${importedTarget})\n")
-  endforeach()
-  set(buildDirExternalPkgsDir
-    "${${PROJECT_NAME}_BINARY_DIR}/${${PROJECT_NAME}_BUILD_DIR_EXTERNAL_PKGS_DIR}")
-  set(tplConfigFile
-    "${buildDirExternalPkgsDir}/${tplName}/${tplName}Config.cmake")
-  file(WRITE "${tplConfigFile}" "${configFileStr}")
-
+    "find_dependency(${externalPkg})\n\n")
+  set(${configFileStrVarInOut} "${configFileStr}" PARENT_SCOPE)
 endfunction()
 #
 # NOTE: Above, ${externalPkg}_DIR is only set when
 # find_package(${externalPkg}) finds a package configure file
 # ${externalPkg}Config.cmake and **not** when it uses a
 # Find${externalPkg}.cmake module.  Therefore, there is no reason to set
 # ${externalPkg}_DIR in this file if it will not be used.
+
+
+function(tribits_extpkg_append_package_config_file_header_with_imported_targets_str
+    tplName  externalPkg  configFileStrVarInOut
+  )
+  set(configFileStr "${${configFileStrVarInOut}}")
+  string(APPEND configFileStr
+    "# TriBITS-compliant Package config file for external package/TPL '${tplName}'\n"
+    "# based on non fully TriBITS-compliant external package '${externalPkg}' that uses\n"
+    "# modern IMPORTED targets\n"
+    "#\n"
+    "# Generated by CMake, do not edit!\n"
+    "\n"
+    "# Guard against multiple inclusion\n"
+    "if (TARGET ${tplName}::all_libs)\n"
```

Comment:
I tried that at one point and it did not behave very well.  Checking for the existence of a target is what is used by the CMake-generated package config file using the export() command so I figured that is the preferred way to avoid multiple definition of the same targets.

But this is an implementation detail so if we find that this needs to be tweaked, that can be done later.